harness/determined/pytorch/_pytorch_trial.py

```diff
@@ -268,75 +541,82 @@ def cleanup_iterator() -> None:
-                        self.context.distributed.size > 1
-                        and not self.context._average_training_metrics
-                    ):
-                        warnings.warn(
-                            "Only the chief worker's training metrics are being reported, due "
-                            "to setting average_training_metrics to False.",
-                            UserWarning,
-                        )
-                elif w.kind == workload.Workload.Kind.COMPUTE_VALIDATION_METRICS:
-                    action = "validation"
-                    response = {
-                        "metrics": self._compute_validation_metrics(),
-                        "stop_requested": self.context.get_stop_requested(),
-                    }
-                elif w.kind == workload.Workload.Kind.CHECKPOINT_MODEL:
-                    action = "checkpointing"
-                    uuid = ""
-                    if self.is_chief:
-                        metadata = {
-                            "determined_version": det.__version__,
-                            "steps_completed": self.steps_completed,
-                            "framework": f"torch-{torch.__version__}",
-                            "format": "pickle",
-                        }
-                        with self.context._core.checkpoint.store_path(metadata) as (
-                            path,
-                            storage_id,
-                        ):
-                            self._save(path)
-                            uuid = storage_id
-                        response = {"uuid": storage_id}
-                    else:
-                        response = {}
-                    uuid = self.context.distributed.broadcast(uuid)
-                    for callback in self.callbacks.values():
-                        callback.on_checkpoint_upload_end(uuid=uuid)
+        return train_unit
 
-                else:
-                    raise AssertionError("Unexpected workload: {}".format(w.kind))
+    def _run(self):
+        min_checkpoint_period = self.env.experiment_config.get_min_checkpoint_period()
+        min_validation_period = self.env.experiment_config.get_min_validation_period()
 
-            except det.InvalidHP as e:
-                logging.info(f"Invalid hyperparameter exception during {action}: {e}")
-                response = workload.InvalidHP()
-            response_func(response)
-            self.upload_tb_files()
+        for op in self.core_context.searcher.operations():
+            self.searcher_op = op
+
+            train_unit = self._get_train_unit_from_op(op)
+
+            min_checkpoint_period = self.as_batches(**min_checkpoint_period)
+            min_validation_period = self.as_batches(**min_validation_period)
+            if min_checkpoint_period < 1:
+                min_checkpoint_period = sys.maxsize
+
+            if min_validation_period < 1:
+                min_validation_period = sys.maxsize
```

Comment:
do this sort of logic outside the for loop, where you read `min_*_period` from the config.